subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -659,6 +663,16 @@ int nrf_cloud_decode_data_endpoint(const struct nrf_cloud_data *input,
 		return err;
 	}
 
+	/* Populate batch endpoint topic by copying and appending /batch to the parsed
+	 * tx endpoint (d2c) topic.
+	 */
+	size_t batch_ep_len_temp = tx_endpoint->len + sizeof(JSON_KEY_TOPIC_BATCH);
+
+	batch_endpoint->ptr = nrf_cloud_calloc(batch_ep_len_temp, 1);
```

Comment:
You should verify if the allocation was successful before using the buffer.

Reply:
Good catch, thanks!